contracts/strategies/LPOptimiserStrategy.sol

```diff
@@ -0,0 +1,270 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity 0.8.9;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "../interfaces/vaults/IERC20Vault.sol";
+import "../interfaces/vaults/IVoltzVault.sol";
+import "../utils/DefaultAccessControl.sol";
+import "../interfaces/utils/ILpCallback.sol";
+import "../libraries/external/FixedPoint96.sol";
+
+contract LPOptimiserStrategy is DefaultAccessControl, ILpCallback {
+    using SafeERC20 for IERC20;
+    using PRBMathUD60x18 for uint256;
+
+    // VAULT PARAMETERS
+    struct VaultParams {
+        int256 sigmaWad; // standard deviation parameter in wad 10^18
+        int256 maxPossibleLowerBoundWad; // Maximum Possible Fixed Rate Lower bounds when initiating a rebalance
+        uint256 proximityWad; // closeness parameter in wad 10^18
+        uint256 weight; // weight parameter that decides how many funds are going to this vault
+    }
+
+    // IMMUTABLES
+    address[] public _tokens;
+    IERC20Vault public immutable _erc20Vault;
+
+    // INTERNAL STATE
+    IVoltzVault[] internal _vaults;
+    VaultParams[] internal _vaultParams;
+    uint256 _totalWeight; // sum of all vault weights
+
+    // CONSTANTS
+    int256 internal constant MINIMUM_FIXED_RATE = 1e16;
+    uint256 internal constant LOG_BASE = 1000100000000000000;
+
+    // GETTERS AND SETTERS
+
+    /// @notice Get the addresses of all vaults
+    function getVaults() public view returns (IVoltzVault[] memory) {
+        return _vaults;
+    }
+
+    /// @notice Get the parameters of a vault 
+    /// @param index The index of the vault in _vaults
+    function getVaultParams(uint256 index) public view returns (VaultParams memory) {
+        return _vaultParams[index];
+    }
+
+    /// @notice Set the parameters of a vault
+    /// @param index The index of the vault in _vaults
+    /// @param vaultParams_ The new parameters of the vault
+    function setVaultParams(uint256 index, VaultParams memory vaultParams_) external {
```

Comment:
Add checks for `vaultParams_` structure fields. For example: `MINIMUM_FIXED_RATE <= vaultParams_.maxPossibleLowerBoundWad`.

Reply:
The parameters are not that constrained and the function is admin-gated.